### PR TITLE
Add `analyze_patients_exposure_in_cols`

### LINKED ISSUE #915: Create analyze counterpart for `summarize_patients_exposure_in_cols`
### Feature description

Create an analyze function `analyze_patients_exposure_in_cols` that will generate data rows instead of content rows so that pagination works correctly.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by edelarua**
Closes #915

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -12,6 +12,9 @@
 * Corrected ordering of factor levels by `d_onco_rsp_label`.
 * Added new "Analyze Functions" and "Summarize Functions" overview pages.
 * Moved all KM plot documentation to the `g_km` function.
+* Added new analyze function `analyze_patients_exposure_in_cols` and analysis function
+  `a_count_patients_sum_exposure` to be used by this analyze function and
```
Comment (Melkiades) [inline NEWS.md:16]:
```suggestion
  `a_count_patients_sum_exposure` for this analyze function and 
```

File: NEWS.md [hunk 2]
```diff
@@ -12,6 +12,9 @@
 * Corrected ordering of factor levels by `d_onco_rsp_label`.
 * Added new "Analyze Functions" and "Summarize Functions" overview pages.
 * Moved all KM plot documentation to the `g_km` function.
+* Added new analyze function `analyze_patients_exposure_in_cols` and analysis function
+  `a_count_patients_sum_exposure` to be used by this analyze function and 
+  `summarize_patients_exposure_in_cols`.
```
Comment (Melkiades) [inline NEWS.md:17]:
```suggestion
* Added `a_count_patients_sum_exposure` for new analyze function `analyze_patients_exposure_in_cols` and `summarize_patients_exposure_in_cols`.
```

File: R/summarize_patients_exposure_in_cols.R [hunk 3]
```diff
@@ -136,28 +179,87 @@ summarize_patients_exposure_in_cols <- function(lyt, # nolintr
...
+#'   summarize_patients_exposure_in_cols(
+#'     var = "AVAL", col_split = TRUE,
+#'     .stats = "n_patients", custom_label = "some custom label"
+#'   ) %>%
+#'   analyze_patients_exposure_in_cols(var = "SEX", col_split = FALSE)
+#' result4 <- build_table(lyt4, df = df, alt_counts_df = adsl)
+#' result4
+#'
+#' lyt5 <- basic_table() %>%
+#'   summarize_patients_exposure_in_cols(var = "AVAL", col_split = TRUE)
```
Comment (Melkiades) [inline R/summarize_patients_exposure_in_cols.R:234]: Maybe here you meant analyze?

File: R/summarize_patients_exposure_in_cols.R [hunk 4]
```diff
@@ -136,28 +179,87 @@ summarize_patients_exposure_in_cols <- function(lyt, # nolintr
...
+#' result5 <- build_table(lyt5, df = df, alt_counts_df = adsl)
+#' result5
+#'
+#' lyt6 <- basic_table() %>%
+#'   summarize_patients_exposure_in_cols(var = "AVAL", col_split = TRUE, .stats = "sum_exposure")
```
Comment (Melkiades) [inline R/summarize_patients_exposure_in_cols.R:239]: also here?

**Review by Melkiades — APPROVED** [review]
Lgtm! Thank you a lot Emily :). Only check if the examples are well representative (e.g. not too many repetitions).

File: R/summarize_patients_exposure_in_cols.R (hunk 3 above)
Comment (shajoezhu) [reply R/summarize_patients_exposure_in_cols.R:234]: do you meant @Melkiades 
```suggestion
#'   summarize_patients_exposure_in_cols(var = "AVAL", col_split = TRUE) %>%
#'   analyze_patients_exposure_in_cols(var = "SEX", col_split = FALSE)
```

Comment (shajoezhu) [reply R/summarize_patients_exposure_in_cols.R:234]: or did you mean? @Melkiades 

```
> lyt5 <- basic_table() %>%
+     analyze_patients_exposure_in_cols(var = "SEX", col_split = TRUE)
> result5 <- build_table(lyt5, df = df, alt_counts_df = adsl)
> result5

         Patients    Person time
————————————————————————————————
Female   6 (50.0%)       46     
Male     6 (50.0%)       68 
```

Comment (clarkliming) [reply R/summarize_patients_exposure_in_cols.R:234]: the naming, probably can be more consistent with other functions. In other functions, `var` is the variable being analyzed, but here it is rather a "split" variable?

**Comment by clarkliming** [comment]
I tried with the following layout
```{R}
basic_table() %>%
  summarize_patients_exposure_in_cols(var = "AVAL", col_split = TRUE) %>%
  #analyze_patients_exposure_in_cols(col_split = TRUE) %>%
  analyze_patients_exposure_in_cols(var = "AVALCAT1", col_split = TRUE) %>%
  build_table(adex, adsl)
```

trying to create both the overall line and the splitted line. went into issue like error in evaluating the argument 'obj' in selecting a method for function 'spl_payload': subscript out of bounds

File: R/summarize_patients_exposure_in_cols.R (hunk 3 above)
Comment (shajoezhu) [reply R/summarize_patients_exposure_in_cols.R:234]: hey @clarkliming , but for this function it was explicitly looking for exposure variable, which is defined by ex_var, var is used only for splitting rows

**Comment by shajoezhu** [comment]
hi @clarkliming , use "col_split = FALSE" will work 

```
> adsl = scda.2022::rcd_2022_01_28_adsl
> adex = scda.2022::rcd_2022_01_28_adex
> basic_table() %>%
+     summarize_patients_exposure_in_cols(var = "AVAL", col_split = TRUE) %>%
+     #analyze_patients_exposure_in_cols(col_split = ) %>%
+     analyze_patients_exposure_in_cols(var = "SEX", col_split = FALSE) %>%
+     build_table(df, adsl)
                                     Patients    Person time
————————————————————————————————————————————————————————————
Total patients numbers/person time   12 (3.0%)       114    
  Female                             6 (1.5%)        46     
  Male                               6 (1.5%)        68  
```

**Comment by clarkliming** [comment]
> hi @clarkliming , use "col_split = FALSE" will work
> 
> ```
> > adsl = scda.2022::rcd_2022_01_28_adsl
> > adex = scda.2022::rcd_2022_01_28_adex
> > basic_table() %>%
> +     summarize_patients_exposure_in_cols(var = "AVAL", col_split = TRUE) %>%
> +     #analyze_patients_exposure_in_cols(col_split = ) %>%
> +     analyze_patients_exposure_in_cols(var = "SEX", col_split = FALSE) %>%
> +     build_table(df, adsl)
>                                      Patients    Person time
> ————————————————————————————————————————————————————————————
> Total patients numbers/person time   12 (3.0%)       114    
>   Female                             6 (1.5%)        46     
>   Male                               6 (1.5%)        68  
> ```

thank you. I tried and it works well!

**Review by shajoezhu — APPROVED** [review]
Thanks so much! @edelarua for the fast turn around!

**Pull request merged**